### PR TITLE
RNA_quant single strand + fastp

### PR DESCRIPTION
**Pull request by hsun3163**
1.Implement what is proposed in #233 
2. Use fastp for adapter removal
3. accommodate single strands

### CODE REVIEW

File: code/molecular_phenotypes/calling/RNA_calling.ipynb [hunk 1]
```diff
@@ -183,7 +183,7 @@
     "    --STAR-index reference_data/STAR_Index/ \\\n",
     "    --gtf reference_data/Homo_sapiens.GRCh38.103.chr.reformatted.gene.ERCC.gtf \\\n",
     "    --container container/rna_quantification.sif \\\n",
-    "    --mem 40G"
+    "    --mem 40G -n"
```
Comment (gaow) [inline code/molecular_phenotypes/calling/RNA_calling.ipynb:186]: why `-n` here? please clean it up if that's not the intention

Comment (hsun3163) [reply code/molecular_phenotypes/calling/RNA_calling.ipynb]: cleaned

File: code/molecular_phenotypes/calling/RNA_calling.ipynb [hunk 2]
```diff
@@ -399,7 +400,9 @@
     "        raise ValueError(\"Duplicated files are found (but should not be allowed) in fastq file list\")\n",
     "    return names, files\n",
     "\n",
-    "sample_id, fastq = get_samples(samples, data_dir)"
+    "sample_id, fastq = get_samples(samples, data_dir)\n",
+    "# Is the RNA-seq data pair-end\n",
+    "is_paired_end = False  if  len(fastq) == len(sample_id) else True "
```
Comment (gaow) [inline code/molecular_phenotypes/calling/RNA_calling.ipynb:405]: I suggest print a message here 

```
from sos.utils import env
env.logger.info('Input samples are `f{"paired-end" if is_paired_end else "single-end"}` sequences.')
```

File: code/molecular_phenotypes/calling/RNA_calling.ipynb [hunk 3]
```diff
@@ -431,14 +434,52 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c2e924a4-fcc0-4f0d-91dc-f5183c09261a",
+   "metadata": {
+    "kernel": "SoS"
+   },
+   "outputs": [],
+   "source": [
+    "[rnaseqc_call_1, rsem_call_1, remove_adaptor]\n",
```
Comment (gaow) [inline code/molecular_phenotypes/calling/RNA_calling.ipynb:454]: Please document in detail what you learned about `fastp` and the default choice of parameters including references to web resources. You can follow the `trimmomatic` section for the level of details needed.

Comment (gaow) [reply code/molecular_phenotypes/calling/RNA_calling.ipynb:454]: Using default for most options might be a good idea ...

File: code/molecular_phenotypes/calling/RNA_calling.ipynb [hunk 4]
```diff
@@ -475,7 +520,7 @@
    },
    "outputs": [],
    "source": [
-    "[rnaseqc_call_1, rsem_call_1, remove_adaptor]\n",
+    "#[rnaseqc_call_1, rsem_call_1, remove_adaptor]\n",
```
Comment (gaow) [inline code/molecular_phenotypes/calling/RNA_calling.ipynb:523]: Please change this line to `[trimmomatic]`

File: code/molecular_phenotypes/calling/RNA_calling.ipynb (hunk 3 above)
Comment (gaow) [inline code/molecular_phenotypes/calling/RNA_calling.ipynb:454]: Please change this line to `[rnaseqc_call_1, rsem_call_1, fastp_remove_adaptor]`

**Comment by gaow** [comment]
@hsun3163 thanks it reads very good! I've made some concrete comments. My only additional request is that you document as much as you can for what you have learned about fastp that you believe useful -- while it is still fresh in your head! Please commit and resolve all my comments, and i'll merge after that.

**Comment by hsun3163** [comment]
APEX, tensorQTL, ss_merger output documentation update as discussed.

**Comment by gaow** [comment]
@hsun3163 i'll merge it anyways for others to see but please keep working on resoving my other comments from yesterday (if not already) after I merge

**Comment by hsun3163** [comment]
> @hsun3163 thanks it reads very good! I've made some concrete comments. My only additional request is that you document as much as you can for what you have learned about fastp that you believe useful -- while it is still fresh in your head! Please commit and resolve all my comments, and i'll merge after that.

Those are all done

**Pull request merged**